### PR TITLE
Feature/pov kotlin rewrite lib peripheral logic

### PR DESCRIPTION
**Pull request by ItsBruno**
## Summary

This PR rewrites peripheral lib logic such as models, utility classes, their interfaces etc. from Java to Kotlin

## Changes
- Refactored peripheral lib logic classes to Kotlin
- Adjusted tests to work with Kotlin - most issues were caused by different property access syntax and nullability behaviour. 
- Removed custom Lazy class and its tests and replaced it with native Kotlin lazy functionality.
- Refactored classes such as UpdateInfo and UpdateResult to data classes.
- Adjusted java classes to work with new Kotlin classes for compilation to work (these changes can be ignored as they will soon be replaced by Kotlin anyway)
- Added logic to remove using deprecated `versionCode` property in Android versions >28 (see `ApplicationConfigurationImpl`). The solution for now throws an error if the version code is too large, therefore large version codes are still unsupported. Supporting Major version codes should be discussed and the lib should be refactored to use Long instead of Int if Major is supported.
- Added small **change to queen-of-versions** to support Kotlin changes in prince-of-versions.
- Some properties which could be null are made non nullable, issues with this will be checked for during the refactor of core lib logic.

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [X] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [X] This pull request introduces a **breaking change**.
- Potential issues with property access syntax compared to the previous version.

## Checklist

- [X] I have performed a self-review of my own code.
- [X] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

### CODE REVIEW

File: prince-of-versions/src/main/java/co/infinum/princeofversions/PresenterImpl.java [hunk 1]
```diff
@@ -48,24 +48,24 @@ public void run() {
     @VisibleForTesting
     UpdateResult run(Loader loader, ApplicationConfiguration appConfig) throws Throwable {
         CheckResult result = interactor.check(loader, appConfig);
-        switch (result.status()) {
+        switch (result.getStatus()) {
```
[the remark this answers is not in the document]
Comment (ItsBruno) [reply prince-of-versions/src/main/java/co/infinum/princeofversions/PresenterImpl.java:51]: Commented on this in an above comment.

**Comment by ItsBruno** [comment]
> Just one question to double check: do all the test pass?
Yes, I run the tests after every file refactored.
